gtest/src/mailbox.rs

```diff
@@ -0,0 +1,99 @@
+use crate::{manager::ExtManager, CoreLog, Log, RunResult};
+use codec::Encode;
+use gear_core::message::{Dispatch, DispatchKind, StoredMessage};
+use gear_core::{
+    ids::{MessageId, ProgramId},
+    message::Payload,
+};
+use std::cell::RefCell;
+
+pub struct Mailbox<'a> {
+    manager_reference: &'a RefCell<ExtManager>,
+    program_id: ProgramId,
+}
+
+impl<'a> Mailbox<'a> {
+    pub(crate) fn new(
+        program_id: ProgramId,
+        manager_reference: &'a RefCell<ExtManager>,
+    ) -> Mailbox<'a> {
+        Mailbox {
+            program_id,
+            manager_reference,
+        }
+    }
+
+    pub fn contains(&self, log: Log) -> bool {
+        self.manager_reference
+            .borrow_mut()
```

Comment:
Why mutable borrowing?
```suggestion
            .borrow()
```

Reply:
Agree with immutable borrowing

Reply:
Contains commonly waits key's reference, not ownable object.
And also better to pass here something which can be meant as Log. So the function decl will be:
`pub fn contains<T: Into<Log> + Clone>(&self, log: &T) -> bool`